monai/data/utils.py

```diff
@@ -528,7 +528,7 @@ def is_supported_format(filename: Union[Sequence[str], str], suffixes: Sequence[
         tokens: Sequence[str] = name.split(".")
         passed: bool = False
         for i in range(len(tokens) - 1):
-            if ".".join(tokens[-(i + 2) : -1]) in suffixes:
+            if ".".join(tokens[-(i + 2) : -1]).lower() in suffixes:
```

Comment:
forgot to mention that suffix extraction could be done with: https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.suffixes

Reply:
Hi @wyli ,

I didn't use this API because it can't support connected suffixes, like "nii.gz", it will generate "nii" and "gz".
And it contains a "." in the suffixes.
Thanks.

Reply:
sure the current impl. is not very safe if the path contains `.`

Reply:
What can be a problem if contains "." in path?
Thanks.

Reply:
I have this function to safely split a file path's directory, basename, and extension reliably:

```py
def split_path_ext(path, full_ext=False):
    '''
    For the given path, return the containing directory, filename without extension, and extension. If `full_ext' is
    True, consider everything to the right of the first period as the extension rather than from the last. For example,
    splitPathExt('foo.bar.baz')[2] produces '.baz' whereas splitPathExt('foo.bar.baz',True)[2] produces '.bar.baz'.
    '''
    path, basename = os.path.split(path)

    if full_extand '.' in basename:
        basename, ext = basename.split('.', 1)  # consider everything to the right of the first . as the extension
        ext = '.' + ext
    else:
        basename, ext = os.path.splitext(basename)  # consider everything to the right of the last . as the extension

    return path, basename, ext
```

Reply:
This function will ensure the path won't get split in the middle of a directory name if it has ".". I don't know what pathlib would do exactly but this function would produce ("/base/name.dir","filename",".tar.gz") for input "/base/name.dir/filename.tar.gz".

Reply:
Hi @ericspod ,

Thanks for your suggestion, good utility function!
My current implementation executes below slightly different logic:
For example, the filename is `test.file.nii.gz`, and supported suffixes: `nii` and `nii.gz`.
My implementation will try from the last suffix "gz": `if "gz" in suffixes`, then try "nii.gz".
I think that's different from your function and `pathlib`.
What do you think?

Thanks.

Reply:
what's the expected output of this? `is_supported_format('.nii_files/test.png', suffixes=['.png'])` I get False...
also
```
from pathlib import PurePath
PurePath('test/.nii_files/test.nii.gz').suffixes
>>> ['.nii', '.gz']
```

Reply:
One possibility:
```py
def is_supported_format(filename: Union[Sequence[str], str], suffixes: Sequence[str]) -> bool:
    short_ext = split_path_ext(filename, False)[2].lower()
    long_ext = split_path_ext(filename, True)[2].lower()
    return any(s.lower() in (short_ext, long_ext) for s in suffixes)
```

Reply:
Hi @ericspod and @wyli ,

Thanks very much for you guys' suggestions here.
I updated the PR combining both your suggestions and it truly simplified a lot and fixed corner issues, then I also added unit tests for it.
Could you please help review it again?

Thanks.